File: examples/ruby/list_target_resolvables.rb
#!/usr/bin/ruby

require 'zypp'
include Zypp

z = ZYppFactory::instance.get_zypp
# puts z.class

t = z.initialize_target(Pathname.new("/"))
# puts t.class

r = z.target.resolvables
# puts r.class

r.each do | p |

    # puts p.class
    puts "#{p.kind} #{p.name} #{p.edition.to_s} #{p.arch.to_s}"
    puts "  Summary: #{p.summary}"
    puts "  Size: #{p.size}"
    puts "  Vendor: #{p.vendor}"
    puts "  Buildtime: #{p.buildtime}"

    d = p.dep(Dep.PROVIDES)
    # puts d.class
    d.each do | x |
	# puts x.class
	puts "  Provides: #{x.to_s}"
    end

    puts

end

File: examples/perl/list_target_resolvables.pl
use zypp;

$z = zyppc::ZYppFactory_instance();
$zypp = $z->getZYpp;

$t = $zypp->initializeTarget(zypp::Pathname->new("/"));

my $repoManager = zypp::RepoManager::new();
my $repos = zypp::RepoManager::knownRepositories($repoManager);

foreach $repo ( @{$repos} ) {
	if ( $repo->enabled() ) {
		print("Caching repo: ".$repo->name."\n");
		if ( ! zypp::RepoManager::isCached($repoManager,$repo) ) {
			print("Rebuilding cache for: ".$repo->name."\n");
			zypp::RepoManager::buildCache($repoManager,$repo);
		}
		zypp::RepoManager::loadFromCache($repoManager,$repo);
	}
}

$store = $zypp->pool;

$it_b = $store->cBegin;
$it_e = $store->cEnd;

while ($store->iterator_equal($it_b, $it_e) ne 1){
   $pkg = $store->iterator_value($it_b);
   print $pkg->kind->asString, " ", $pkg->name, " ", $pkg->edition->asString;
   print $pkg->arch->string, "\n";
   print "  Summary: ", $pkg->summary, "\n";
   print "  Size: ", $pkg->installSize->asString, "\n";
   print "  Vendor: ", $pkg->vendor->asString, "\n";
   print "  BuildTime: ", $pkg->buildtime->asString, "\n";
   $it_b = $store->iterator_incr($it_b);

   $deps = $pkg->dep($zyppc::Dep_PROVIDES);

   $it_b2 = $deps->cBegin;

   while($deps->iterator_equal($it_b2, $deps->cEnd) ne 1){
      $dep = $deps->iterator_value($it_b2);
      print "Provides: ", $dep->asString, "\n";
      $it_b2 = $deps->iterator_incr($it_b2);
   }
   print "\n";
   
}
